metrics-delivery-endpoint-impl/src/main/java/eu/openaire/mas/delivery/provider/ExpressionContext.java

```diff
@@ -0,0 +1,52 @@
+package eu.openaire.mas.delivery.provider;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+
+import com.github.anhdat.PrometheusApiClient;
+import com.github.anhdat.models.MatrixResponse;
+
+/**
+ * Context for SpEL expressions executed by PrometheusMetricsProvider.
+ */
+class ExpressionContext {
+    private PrometheusApiClient prometheusClient;
+
+    ExpressionContext(PrometheusApiClient prometheusClient) {
+	this.prometheusClient = prometheusClient;
+    }
+
+    /**
+     * Sums values of a series provided periodically for which there is another series provided
+     * in the same exposition which is different (usually because it represents a timestamp) for each
+     * new value of the series to sum.
+     *
+     * @param countsSeries the series to sum
+     * @param stampsSeries the series of timestamps
+     * @param from initial timestamp to go back to
+     * @return sum of values from the countsSeries matching each distinct value of the stampsSeries
+     */
+    public float sumPeriods(String countsSeries, String stampsSeries, long from) throws IOException {
```

Comment:
Just wondering: wouldn't it be useful to put `to` as a an additional and optional timestamp (current time when missing) we will be accumulating the values until?

I mean I was thinking about the scenario of calculating the total number of API hits for the whole previous year.

Reply:
Yes, it definitely would. But at the moment the API does not handle it yet anyway (#23).
When we add that, we need to handle it here too. I think extending the expression context with the requested timestamp (and then using it in `sumPeriods` and other methods if we have them) will be the way to go.